Demo/Application/Base/AppDelegate.swift

```diff
@@ -0,0 +1,82 @@
+            UserDefaults.standard.set(BraintreeDemoAuthType.clientToken, forKey: BraintreeDemoSettings.AuthorizationTypeDefaultsKey)
+        } else if ProcessInfo.processInfo.arguments.contains("-TokenizationKey") {
+            UserDefaults.standard.set(BraintreeDemoAuthType.tokenizationKey, forKey: BraintreeDemoSettings.AuthorizationTypeDefaultsKey)
+        } else if ProcessInfo.processInfo.arguments.contains("-MockedPayPalTokenizationKey") {
+            UserDefaults.standard.set(BraintreeDemoAuthType.mockedPayPalTokenizationKey, forKey: BraintreeDemoSettings.AuthorizationTypeDefaultsKey)
+        } else if ProcessInfo.processInfo.arguments.contains("-UITestHardcodedClientToken") {
+            UserDefaults.standard.set(BraintreeDemoAuthType.uiTestHardcodedClientToken, forKey: BraintreeDemoSettings.AuthorizationTypeDefaultsKey)
+        }
+        
+        UserDefaults.standard.removeObject(forKey: "BraintreeDemoSettingsAuthorizationOverride")
+        ProcessInfo.processInfo.arguments.forEach { arg in
+            if arg.contains("-Integration:") {
+                let testIntegration = arg.replacingOccurrences(of: "-Integration:", with: "")
+                UserDefaults.standard.setValue(testIntegration, forKey: "BraintreeDemoSettingsIntegration")
+            } else if arg.contains("-Authorization:") {
+                let testIntegration = arg.replacingOccurrences(of: "-Authorization:", with: "")
+                UserDefaults.standard.setValue(testIntegration, forKey: "BraintreeDemoSettingsAuthorizationOverride")
+            }
+        }
+        
+        if ProcessInfo.processInfo.arguments.contains("-ClientTokenVersion2") {
+            UserDefaults.standard.set("2", forKey: "BraintreeDemoSettingsClientTokenVersionDefaultsKey")
+        } else if ProcessInfo.processInfo.arguments.contains("-ClientTokenVersion3") {
+            UserDefaults.standard.set("3", forKey: "BraintreeDemoSettingsClientTokenVersionDefaultsKey")
+        }
+        // End checking for testing arguments
+        
+        guard let settingsBundle = Bundle.main.path(forResource: "Settings", ofType: "bundle"),
+              let settings = NSDictionary(contentsOfFile: settingsBundle.appending("/Root.plist")) else {
+            print("Could not find Settings.bundle")
+            return
+        }
+                
+        if let preferences = settings.object(forKey: "PreferenceSpecifiers") as? Array<[String: Any]> {
+            var defaultsToRegister: [String: Any] = [:]
+            preferences.forEach { prefSpecification in
+                print(prefSpecification)
+                if let key = prefSpecification["Key"] as? String, prefSpecification.keys.contains("DefaultValue") {
+                    defaultsToRegister[key] = prefSpecification["DefaultValue"]
+                }
+            }
+            
+            UserDefaults.standard.register(defaults: defaultsToRegister)
+        }
```

Comment:
I think we can delete these lines. I'm not sure really what they do

Reply:
My assumption was it saves things like selecting client token or the demo type between sessions to user defaults but that may be totally incorrect

Reply:
Yeah it's a bit wonky for sure. I wonder if we get some of this for free if we move away from `InAppSettingsKit`